js_basics_oop/Book.js

```diff
@@ -0,0 +1,44 @@
+export class Book {
+
+    static getOldestBook(books) {
+        if (books.length === 0) {
+            return null
+        }
+        return [...books].sort((a, b) => a._year - b._year)[0];
+    }
+
+    constructor(name, author, year) {
+        this._name = name;
+        this._author = author;
+        this._year = year;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get author() {
+        return this._author;
+    }
+
+    get year() {
+        return this._year;
+    }
+
+    set name(newName) {
+        this._name = typeof newName === "string" ? newName : this._name;
```

Comment:
you reassign this value every time when setter is called. 
Just ignore case when newValue is invalid
```js
if (typeof newName === 'string') this._name = newName;
```

the same for others setters